### PR TITLE
Wired up the shipping zones UI

### PR DESCRIPTION
**Pull request by marcinbot**
* new shipping zone page at `/store/settings/shipping/:site/zone/:zone`
* loading the shipping zones from the server on the general shipping settings page and individual zones page
* adding shipping methods and editing their title, type and method-specific settings
* deleting shipping methods

### CODE REVIEW

File: client/extensions/woocommerce/app/settings/shipping/shipping-zone-list.js [hunk 1]
```diff
@@ -12,72 +14,105 @@ import Card from 'components/card';
...
+
+		const onAddZoneClick = () => ( this.props.addNewShippingZone( siteId ) );
 
 		return (
 			<div>
 				<ExtendedHeader
 					label={ translate( 'Shipping Zones' ) }
 					description={ translate( 'The regions you ship to and the methods you will provide.' ) }>
-					<Button onClick={ onAddZoneOpen }>{ translate( 'Add zone' ) }</Button>
+					<Button onClick={ onAddZoneClick }>{ translate( 'Add zone' ) }</Button>
```
Comment (a8ci18n) [inline client/extensions/woocommerce/app/settings/shipping/shipping-zone-list.js]: Hi! I've found a possible matching string that has already been translated **9** times:
`translate( 'Add DNS Zone' )` *ES Score: 9*
See 1 additional suggestion in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/wired-up-shipping)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

File: client/extensions/woocommerce/state/sites/shipping-zone-methods/selectors.js [hunk 2]
```diff
@@ -30,6 +30,31 @@ export const getShippingZoneMethod = ( state, id, siteId = getSelectedSiteId( st
 
 /**
  * @param {Object} state Whole Redux state tree
+ * @param {Number} zoneId Shipping Zone ID to get shipping methods for
+ * @param {Number} [siteId] Site ID to get. If not provided, the Site ID selected in the UI will be used
+ * @return {Array|null} The shipping zone array, or null if it wasn't found
+ */
+export const getShippingZoneMethods = ( state, zoneId, siteId = getSelectedSiteId( state ) ) => {
```
Comment (DanReyLop) [inline client/extensions/woocommerce/state/sites/shipping-zone-methods/selectors.js]: This function shouldn't be used directly by the UI.

File: client/extensions/woocommerce/app/settings/shipping/shipping-zone-list.js [hunk 3]
```diff
@@ -12,72 +14,87 @@ import Card from 'components/card';
...
+
+		const onAddZoneClick = () => ( this.props.addNewShippingZone( siteId ) );
+
+		return (
+			<div>
 				<ExtendedHeader
 					label={ translate( 'Shipping Zones' ) }
 					description={ translate( 'The regions you ship to and the methods you will provide.' ) }>
-					<Button onClick={ onAddZoneOpen }>{ translate( 'Add zone' ) }</Button>
+					<Button onClick={ onAddZoneClick } disabled={ ! loaded }>{ translate( 'Add zone' ) }</Button>
```
Comment (a8ci18n) [inline client/extensions/woocommerce/app/settings/shipping/shipping-zone-list.js]: Hi! I've found a possible matching string that has already been translated **9** times:
`translate( 'Add DNS Zone' )` *ES Score: 9*
See 1 additional suggestion in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/wired-up-shipping)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

File: client/extensions/woocommerce/app/settings/shipping/shipping-methods/flat-rate.js [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+			currencySymbolPrefix={ '$' }
+			currencySymbolSuffix={ '' }
+			value={ cost }
+			onChange={ onCostChange } />;
+	};
+
+	return (
+		<div>
+			<FormFieldSet>
+				<FormLabel>{ translate( 'Cost:' ) }</FormLabel>
```
Comment (a8ci18n) [inline client/extensions/woocommerce/app/settings/shipping/shipping-methods/flat-rate.js]: Hi! I've found a possible matching string that has already been translated **59** times:
`translate( 'Cost' )` *ES Score: 12*
See 1 additional suggestion in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/wired-up-shipping)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

File: client/extensions/woocommerce/app/settings/shipping/shipping-methods/free-shipping.js [hunk 5]
```diff
@@ -1,68 +1,109 @@
...
+
+	return (
+		<FormFieldSet>
+			<FormLabel>{ translate( 'Who gets free shipping?' ) }</FormLabel>
+			<div className="shipping-methods__free-shipping-option">
+				<FormRadio
+					value=""
+					checked={ ! requires }
+					onChange={ onConditionChange } />
+				{ translate( 'Everyone!' ) }
```
Comment (a8ci18n) [inline client/extensions/woocommerce/app/settings/shipping/shipping-methods/free-shipping.js]: Hi! I've found a possible matching string that has already been translated **35** times:
`translate( 'Everyone', { context: 'Filter label for posts list'} )` *ES Score: 21*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

File: client/extensions/woocommerce/app/settings/shipping/shipping-method-utils.js [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+	}
+};
+
+export const getMethodSummary = ( method ) => {
+	switch ( method.methodType ) {
+		case 'flat_rate':
+			return translate( 'Cost: %s', { args: [ method.cost ] } );
+		case 'free_shipping':
+			if ( ! method.requires ) {
+				return translate( 'For everyone' );
```
Comment (a8ci18n) [inline client/extensions/woocommerce/app/settings/shipping/shipping-method-utils.js]: Hi! I've found a possible matching string that has already been translated **35** times:
`translate( 'Everyone', { context: 'Filter label for posts list'} )` *ES Score: 11*
See 2 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/wired-up-shipping)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

File: client/extensions/woocommerce/app/settings/shipping/shipping-zone-list.js [hunk 7]
```diff
@@ -10,74 +12,102 @@ import { localize } from 'i18n-calypso';
...
 			<div>
 				<ExtendedHeader
 					label={ translate( 'Shipping Zones' ) }
 					description={ translate( 'The regions you ship to and the methods you will provide.' ) }>
-					<Button onClick={ onAddZoneOpen }>{ translate( 'Add zone' ) }</Button>
+					<Button
+						href={ addNewHref }
+						onClick={ onAddNewClick }
+						disabled={ ! loaded }>{
+							translate( 'Add zone' ) }
```
Comment (a8ci18n) [inline client/extensions/woocommerce/app/settings/shipping/shipping-zone-list.js]: Hi! I've found a possible matching string that has already been translated **9** times:
`translate( 'Add DNS Zone' )` *ES Score: 9*
See 1 additional suggestion in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/wired-up-shipping)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

File: client/extensions/woocommerce/app/settings/shipping/shipping-zone/shipping-zone-locations.js [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+	const renderContent = () => {
+		if ( ! loaded ) {
+			return (
+				<div className="shipping-zone__loading-spinner">
+					<Spinner size={ 24 } />
+				</div>
+			);
+		}
+
+		return ( <Button compact>{ translate( 'Add location' ) }</Button> );
```
Comment (a8ci18n) [inline client/extensions/woocommerce/app/settings/shipping/shipping-zone/shipping-zone-locations.js]: Hi! I've found a possible matching string that has already been translated **46** times:
`translate( 'Add Location' )` *ES Score: 15*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

**Comment by kellychoffman** [comment]
I don't want to mess you up and edit this myself, but could you remove the `This is the address you entered while signing up for a WordPress.com Store.` info notice in the Shipping Origin section? Its no longer relevant. Or let me know if I should do this in another PR.

File: client/extensions/woocommerce/app/settings/shipping/shipping-zone/shipping-methods/flat-rate.js [hunk 9]
```diff
@@ -0,0 +1,76 @@
...
+			currencySymbolPrefix={ '$' }
+			currencySymbolSuffix={ '' }
+			value={ cost }
+			onChange={ onCostChange } />;
+	};
+
+	return (
+		<div>
+			<FormFieldSet>
+				<FormLabel>{ translate( 'Cost:' ) }</FormLabel>
```
Comment (a8ci18n) [inline client/extensions/woocommerce/app/settings/shipping/shipping-zone/shipping-methods/flat-rate.js]: Hi! I've found a possible matching string that has already been translated **59** times:
`translate( 'Cost' )` *ES Score: 12*
See 1 additional suggestion in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/wired-up-shipping)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

File: client/extensions/woocommerce/app/settings/shipping/shipping-zone/shipping-methods/free-shipping.js [hunk 10]
```diff
@@ -0,0 +1,109 @@
...
+
+	return (
+		<FormFieldSet>
+			<FormLabel>{ translate( 'Who gets free shipping?' ) }</FormLabel>
+			<div className="shipping-methods__free-shipping-option">
+				<FormRadio
+					value=""
+					checked={ ! requires }
+					onChange={ onConditionChange } />
+				{ translate( 'Everyone!' ) }
```
Comment (a8ci18n) [inline client/extensions/woocommerce/app/settings/shipping/shipping-zone/shipping-methods/free-shipping.js]: Hi! I've found a possible matching string that has already been translated **35** times:
`translate( 'Everyone', { context: 'Filter label for posts list'} )` *ES Score: 21*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

File: client/extensions/woocommerce/app/settings/shipping/shipping-zone/utils.js [hunk 11]
```diff
@@ -0,0 +1,55 @@
...
+	}
+};
+
+export const getMethodSummary = ( method ) => {
+	switch ( method.methodType ) {
+		case 'flat_rate':
+			return translate( 'Cost: %s', { args: [ method.cost ] } );
+		case 'free_shipping':
+			if ( ! method.requires ) {
+				return translate( 'For everyone' );
```
Comment (a8ci18n) [inline client/extensions/woocommerce/app/settings/shipping/shipping-zone/utils.js]: Hi! I've found a possible matching string that has already been translated **35** times:
`translate( 'Everyone', { context: 'Filter label for posts list'} )` *ES Score: 11*
See 2 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/wired-up-shipping)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

File: client/extensions/woocommerce/app/settings/shipping/shipping-zone/shipping-zone-locations.js [hunk 12]
```diff
@@ -0,0 +1,44 @@
...
+	const renderContent = () => {
+		if ( ! loaded ) {
+			return (
+				<div className="shipping-zone__loading-spinner">
+					<Spinner size={ 24 } />
+				</div>
+			);
+		}
+
+		return ( <Button>{ translate( 'Add location' ) }</Button> );
```
Comment (a8ci18n) [inline client/extensions/woocommerce/app/settings/shipping/shipping-zone/shipping-zone-locations.js]: Hi! I've found a possible matching string that has already been translated **46** times:
`translate( 'Add Location' )` *ES Score: 15*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

File: client/extensions/woocommerce/app/settings/shipping/shipping-zone/shipping-zone-locations.js [hunk 13]
```diff
@@ -0,0 +1,51 @@
...
+				<div className="shipping-zone__loading-spinner">
+					<Spinner size={ 24 } />
+				</div>
+			);
+		}
+
+		return (
+			<ListItem>
+				<ListItemField>
+					<Button>{ translate( 'Add location' ) }</Button>
```
Comment (a8ci18n) [inline client/extensions/woocommerce/app/settings/shipping/shipping-zone/shipping-zone-locations.js]: Hi! I've found a possible matching string that has already been translated **46** times:
`translate( 'Add Location' )` *ES Score: 15*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

**Comment by kellychoffman** [comment]
Instead of the spinner, can we show a loading indicator like in Payments? 

screen shot 2017-06-19 at 4 21 03 pm

**Comment by kellychoffman** [comment]
When adding a method, instead of automatically adding it to the list and having to press the 'edit' button, can the modal pop up? Its nice to review everything in there before adding it.

**Pull request closed without merging**